implementations/micrometer-registry-datadog/src/main/java/io/micrometer/datadog/DatadogMeterRegistry.java

```diff
@@ -78,17 +82,63 @@ public DatadogMeterRegistry(DatadogConfig config, Clock clock) {
      */
     @Deprecated
     public DatadogMeterRegistry(DatadogConfig config, Clock clock, ThreadFactory threadFactory) {
-        this(config, clock, threadFactory, new HttpUrlConnectionSender(config.connectTimeout(), config.readTimeout()));
+        this(config, clock, threadFactory, new HttpUrlConnectionSender(config.connectTimeout(), config.readTimeout()),
+                null);
+    }
+
+    /**
+     * @param config Configuration options for the registry that are describable as
+     * properties.
+     * @param clock The clock to use for timings.
+     * @param threadFactory The thread factory to use to create the publishing thread.
+     * @deprecated Use {@link #builder(DatadogConfig)} instead.
+     */
+    @Deprecated
+    public DatadogMeterRegistry(DatadogConfig config, Clock clock, ThreadFactory threadFactory,
+            @Nullable StatsDClient customClient) {
+        this(config, clock, threadFactory, new HttpUrlConnectionSender(config.connectTimeout(), config.readTimeout()),
+                customClient);
     }
 
-    private DatadogMeterRegistry(DatadogConfig config, Clock clock, ThreadFactory threadFactory,
-            HttpSender httpClient) {
+    /**
+     * Truthy if the uri is defined for a dogstatsd reporting configuration.
+     * @return Returns true if this is a non https scheme target, to indicate to talk via
+     * dogstatsd.
+     */
+    static boolean isStatsd(String uri) {
+        String scheme = URI.create(uri).getScheme();
+        return !(scheme.equalsIgnoreCase("https") || scheme.equalsIgnoreCase("http"));
+    }
+
+    @SuppressWarnings({ "Var", "Varifier" })
+    private DatadogMeterRegistry(DatadogConfig config, Clock clock, ThreadFactory threadFactory, HttpSender httpClient,
+            @Nullable StatsDClient statsDClient) {
         super(config, clock);
 
         config().namingConvention(new DatadogNamingConvention());
 
+        if (statsDClient == null && isStatsd(config.uri())) {
+            NonBlockingStatsDClientBuilder builder = new NonBlockingStatsDClientBuilder();
+            URI statsdURI = URI.create(config.uri());
+            if (statsdURI.getScheme().equalsIgnoreCase("tcp") || statsdURI.getScheme().equalsIgnoreCase("udp")) {
+                if (!statsdURI.getHost().equalsIgnoreCase("")) {
```

Comment:
We could theoretically flip the equality - in case any element here is `null` (although that shouldn't really happen)